packages/odyssey/CHANGELOG.md

```diff
@@ -5,6 +5,29 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.4.0] - 2020-07-31
+
+### Added
+
+- `@okta/odyssey-icons` package: includes an initial set of SVGs and process for batch optimizing via `svgo`
+- `@okta/vuepress-theme-nimatron` package: supplies the default theme for Odyssey's docs
```

Comment:
This can probably be removed as well - since there are no immediate plans to make the theme package public.

Reply:
@edburyenegren-okta - Since you want to keep dev-related changes in the CHANGELOG, common convention is to mark these as `internal`. Given the pattern this repo adopted - the `[internal]` label would be a clean solution for this. That way there is clear separation between what is actionable for public vs internal devs:


```suggestion
- [internal] `@okta/vuepress-theme-nimatron` package: supplies the default theme for Odyssey's docs
```

Reply:
KeepAChangelog advocates against this (it's been an open issue there since 2014):

> While I have used [INTERNAL] tags (on individual changes as @bramdevries demonstrated), it was to mark potentially irrelevant changes to end-users in order to parse out a different version of the changelog and use it as release notes.
> 
> After doing this for a few months I realized that release notes and change logs have purposes but ultimately different audiences (people using software vs. people depending on software) and transparency requirements when it comes to communication: **a change log needs to disclose any relevant changes because the most insignificant change may introduce a regression, while the same insignificant change may introduce a bug it's unlikely that it will ever be communicated in release notes.**
> - _https://github.com/olivierlacan/keep-a-changelog/issues/30_

I'm onboard with Olivier's rationale here (changelogs should capture all any change that could lead to a regression) - especially for an OSS project - and would prefer to stick to the spec unless there's a separate one we can adopt.

Reply:
I think we're interpreting his argument differently. He states:

> I'm not convinced (yet) that [INTERNAL] changes belong in a change log but if a majority agrees that it's a good idea to extract out a public version that filters out [INTERNAL]changes (see #51) then I'm more than fine with that.

You called out earlier that the audience of this CHANGELOG is both consumers and the active development team. Given that, there is justification for marking these as `internal` changes. For the [Sign-in Widget](https://github.com/okta/okta-signin-widget/releases), they utilize an **Other** section to callout changes like this.

External developers who typically read CHANGELOG entries do a few things:
1. Search for anything marked as a **BREAKING CHANGE** (things that lead to regressions).
2. Look for any notable changes that'll potentially effect their consumption of this library.
3. Examine new features

If your target is for **external developers** - we should remove all internal changes. Since it isn't - I think there is an excellent use-case for adding an `[internal]` label.

Reply:
If you follow the whole thread and link, his argument is for using `[Internal]` as a flag when programmatically creating two changelogs - one public, and one internal, not for identifying those changes within a log. If you only have one changelog, those changes affect all users whether they are interested in the changes or not.

That is, _anyone_ who wants to build the docs locally needs to know the things included here: that something was added to a `.gitignore`, that our docs theme is sourced from a local package, that our docs framework is an entirely different piece of software now. It doesn't matter if they are an Okta dev, an Okta designer, a customer dev, or a random third-party contributor (which we get from time to time already). Their persona or position do not necessarily determine their use case, and it doesn't determine whether a regression will occur or how to troubleshoot said regression.

Since this issue has been open for multiple years and engaged with by the maintainer, the most direct solve for us is to either trust the rationale and stick to the spec or move to a different standard if one exists. Homebrewing changelog ideology doesn't seem efficient.

Reply:
A CHANGELOG entry does not solve the use case you're describing in relation to docs. There is no documentation here, under `docs`, or within `vuepress-theme-nimatron` that explains how these are related. To me, what you're describing needs to be outlined in the [upgrading notes](https://github.com/olivierlacan/keep-a-changelog/issues/34). As a result, this is a bulleted item listing a **feature** without context.

I agree 💯 with your statement that audience shouldn't matter. They're intended to be readable by **humans**. Where we don't see eye-to-eye is what is classified as a CHANGLOG entry versus what can just be a commit into source.

As a middle ground, what do you think about moving this "internal-like" changes to the **CHANGED** section:

```diff
### Changed
...
+ - Changed the Odyssey documentation platform from `hexo` to `vuepress`.
+    - Extract the docs theme into a new package:  `@okta/vuepress-theme-nimatron` (see README for setup instructions).
+    - `.gitignore` now includes the auto-generated `CHANGELOG` to avoid creating duplicate files.
```

That way we satisfy a few things:
1. The important aspect of the change introduced in `0.4.0` is outlined in our CHANGELOG.
2. Readers now view this as a **change** instead of an **addition**.
3. Valuable documentation is provided for those wishing to contribute to the repo.

Reply:
This works for me - updated!

I think we are also running into some of the friction that maintaining a single changelog/version for multiple packages comes with. The more packages we have, the more changes will feel unimportant when surfaced to a single log. Problem for another day though!